### ISSUE TITLE
string_view and VS 2017

### ISSUE THREAD

**Issue by jcelerier** [chriskohlhoff/asio#230]
Just including these two : 

```C++
#include "asio/detail/config.hpp"
#include "asio/detail/string_view.hpp"
```

fails with 
```
#error directive: string_view is only available with C++17
```
(i.e. one has to build with /std:c++17 or maybe /std:c++latest, which is not always possible: for instance currently this breaks boost due to the use of deprecated `std::unary_function` stuff in it).
 
Interestingly, including either one or the other works, as well as changing the order of the includes.

```C++
// ok: 
#include "asio/detail/string_view.hpp"
#include "asio/detail/config.hpp"
```

a solution could be to guard the `#include <string_view>` with `defined(_HAS_CXX17)` since it seems that's the macro MS uses in their string_view header.

**Comment by chriskohlhoff** [comment]
Fixed in d6144828beb342548eaf7b31132217247cb03e27.

**Comment by jcelerier** [comment]
thanks :)
